compiler-core/src/build/package_compiler.rs

```diff
@@ -105,68 +113,74 @@ where
             return e.into();
         }
 
-        let artefact_directory = self.out.join(paths::ARTEFACT_DIRECTORY_NAME);
-        let codegen_required = if self.perform_codegen {
-            CodegenRequired::Yes
+        let outcome = if self.skip_analysis {
```

Comment:
There's lots of nesting here and the second branch is very long, making it hard to read. How about this instead?

```rust
// some docs here
let loaded = if self.compile_modules {
  loader.run()
} else {
  Loaded::empty()
};
```

There's more setup done in the case when it's not used, but I think it'd be easier to follow.

Reply:
Yeah agreed!